File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> result;
        sort(intervals.begin(), intervals.end());
        result.push_back(intervals[0]);
        int resultIndex = 0;
        for (int i=1; i<intervals.size(); i++){
            if (result[resultIndex][1] >= intervals[i][0]){
                result[resultIndex][1] = max(result[resultIndex][1], intervals[i][1]);
            }
            else{
                result.push_back(intervals[i]);
                resultIndex++;
            }
        }
        return result;

}
};

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution(object):
    def merge(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: List[List[int]]
        """
        sortedIntervals = sorted(intervals)
        out = []
        i=0
        while (i<len(intervals)):
            small = sortedIntervals[i][0]
            big = sortedIntervals[i][1]
            while(i+1 < len(intervals)):
                next_small = sortedIntervals[i+1][0]
                next_big = sortedIntervals[i+1][1]
                if small <= next_big and big >= next_small:
                    big = max(big, next_big)
                    small = min(small, next_small)
                    i += 1
                else:
                    break
            out.append([small, big])
            i += 1

        return out
                



        